### PR TITLE
Do not permit the use of a modified 'main.dol' file

### PR DESCRIPTION
**Pull request by MikeIsAStar**

### CODE REVIEW

**Comment by MikeIsAStar** [comment]
This pull request is now ready for review @stblr.

File: configure.py [hunk 1]
```diff
@@ -1119,11 +1119,13 @@
         os.path.join('common', 'FS.cc'),
         os.path.join('common', 'ICache.cc'),
         os.path.join('common', 'IOS.cc'),
+        os.path.join('common', 'Memcmp.c'),
         os.path.join('common', 'Memcpy.c'),
         os.path.join('common', 'Memset.c'),
         os.path.join('common', 'Strlcpy.c'),
         os.path.join('common', 'Strlen.c'),
         os.path.join('common', 'VI.cc'),
+        os.path.join('vendor', 'clibs', 'sha1', 'sha1.c'),
```
Comment (stblr) [inline configure.py:1128]: What's the purpose of the `clibs` directory?

File: loader/Loader.cc [hunk 2]
```diff
@@ -9,6 +11,10 @@
 #include <common/Paths.hh>
 #include <common/VI.hh>
 
+extern "C" {
+#include <vendor/clibs/sha1/sha1.h>
+}
```
Comment (stblr) [inline loader/Loader.cc:16]:
```suggestion
extern "C" {
#include <vendor/clibs/sha1/sha1.h>
}
```

File: loader/Dol.cc [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            {
+                0xFF, 0x41, 0xE4, 0x22, 0x20, 0x0C, 0x2F, 0x37, 0x94, 0x29,
+                0xC3, 0x48, 0x99, 0x3D, 0x21, 0xEB, 0xE7, 0x6E, 0x61, 0xE2,
+            },
+            0x00373160,
+        },
+        // clang-format on
+};
+
+static std::optional<u32> GetRegionIndex() {
```
Comment (stblr) [inline loader/Dol.cc:57]: We should try to use a consistent region order...

File: loader/Loader.cc [hunk 4]
```diff
@@ -1,5 +1,7 @@
 #include "Apploader.hh"
 
+#include "Dol.hh"
```
Comment (stblr) [inline loader/Loader.cc:3]:
```suggestion
#include "loader/Dol.hh"
```

File: loader/Apploader.cc [hunk 5]
```diff
@@ -134,6 +138,11 @@ std::optional<GameEntryFunc> LoadAndRun(IOS::DI &di) {
 
     init(report);
 
+    std::optional<u32> dolSize = Dol::GetSize();
+    if (dolSize.has_value()) {
+        memset((void *)Dol::GetStartAddress(), 0, *dolSize);
```
Comment (stblr) [inline loader/Apploader.cc:143]: `const_cast`?

File: loader/Apploader.cc [hunk 6]
```diff
@@ -1,9 +1,13 @@
 #include "Apploader.hh"
 
+#include "Dol.hh"
```
Comment (stblr) [inline loader/Apploader.cc:3]:
```suggestion
#include "loader/Dol.hh"
```

**Review by stblr — CHANGES_REQUESTED** [review]

File: configure.py (hunk 1 above)
Comment (MikeIsAStar) [reply configure.py]: Resolved in commit [0c18826](https://github.com/stblr/mkw-sp/pull/659/commits/0c18826c45a999dbfdbc3268248d724d5700441f).

File: loader/Loader.cc (hunk 2 above)
Comment (MikeIsAStar) [reply loader/Loader.cc]: Resolved in commit [0c18826](https://github.com/stblr/mkw-sp/pull/659/commits/0c18826c45a999dbfdbc3268248d724d5700441f).

File: loader/Dol.cc (hunk 3 above)
Comment (MikeIsAStar) [reply loader/Dol.cc:57]: Resolved in commit [0c18826](https://github.com/stblr/mkw-sp/pull/659/commits/0c18826c45a999dbfdbc3268248d724d5700441f).

File: loader/Apploader.cc (hunk 5 above)
Comment (MikeIsAStar) [reply loader/Apploader.cc]: Resolved in commit [0c18826](https://github.com/stblr/mkw-sp/pull/659/commits/0c18826c45a999dbfdbc3268248d724d5700441f).

File: loader/Apploader.cc (hunk 6 above)
Comment (MikeIsAStar) [reply loader/Apploader.cc]: Resolved in commit [0c18826](https://github.com/stblr/mkw-sp/pull/659/commits/0c18826c45a999dbfdbc3268248d724d5700441f).

**Review by stblr — APPROVED** [review]

**Comment by Maidvelia** [comment]
If main.dol cannot be edited then a separate option to disable the deflicker should be put in.

**Pull request merged**